src/gluonts/model/n_beats/_ensemble.py

```diff
@@ -0,0 +1,427 @@
+        with (path / "parameters.json").open("r") as fp:
+            parameters = load_json(fp.read())
+
+        # basically save each predictor in its own sub-folder
+        num_predictors = parameters["num_predictors"]
+        num_digits = len(str(num_predictors))
+
+        predictors = []
+        # load all the predictors individually and also make sure not to load anything else by mistake
+        predictor_locations = [
+            f"predictor_{str(index).zfill(num_digits)}"
+            for index in range(num_predictors)
+        ]
+
+        # deserialize predictors
+        for sub_dir in predictor_locations:
+            predictors.append(
+                RepresentableBlockPredictor.deserialize(path / sub_dir, ctx)
+            )
+
+        return NBEATSEnsemblePredictor(
+            prediction_length=parameters["prediction_length"],
+            freq=parameters["freq"],
+            predictors=predictors,
+            aggregation_method=parameters["aggregation_method"],
+        )
+
+    # Additionally implemented since we are dealing with RepresentableBlockPredictor
+    def hybridize(self, batch: DataBatch) -> None:
+        for predictor in self.predictors:
+            predictor.hybridize(batch)
+
+    def predict(
+        self, dataset: Dataset, num_samples: Optional[int] = 1, **kwargs
+    ) -> Iterator[Forecast]:
+        if num_samples != 1:
+            logging.warning(
+                "NBEATSEnsemblePredictor does not support sampling. "
+                "Therefore 'num_samples' will be ignored and set to 1."
+            )
+        iterators = []
+
+        # create the iterators from the predictors
+        for predictor in self.predictors:
+            iterators.append(predictor.predict(dataset, num_samples=1))
+
+        # we always have to predict for one series in the dataset with
+        # all models and return it as a 'SampleForecast' so that it is
+        # clear that all these prediction concern the same series
+        for item in dataset:
+            output = []
+            start_date = None
+
+            for iterator in iterators:
+                prediction = next(iterator)
+                output.append(prediction.samples)
+
+                # get the forecast start date
+                if start_date is None:
+                    start_date = prediction.start_date
+            output = np.stack(output, axis=0)
+
+            # aggregating output of different models
+            # default according to paper is median,
+            # but we can also make use of not aggregating
+            if self.aggregation_method == "median":
+                output = np.median(output, axis=0)
+            elif self.aggregation_method == "mean":
+                output = np.mean(output, axis=0)
+            else:  # "none": do not aggregate
+                pass
+
+            yield SampleForecast(
+                output,
+                start_date=start_date,
+                freq=start_date.freqstr,
```

Comment:
You can assert that `start_date is not None` before the get rid of mypy's complaint (https://ci.mxnet.io/blue/rest/organizations/jenkins/pipelines/GluonTS-py3-cpu-unittest/branches/PR-553/runs/9/nodes/42/steps/66/log/?start=0)